revamp MPIPreferences docs (#594)
JuliaParallel/MPI.jl: docs/src/configuration.md

system-provided MPI implementation in order to exploit features such as fast network
interfaces and CUDA-aware MPI interfaces.

> This can be configured using the MPIPreferences.jl package, which allows the user to choose their MPI implementation. This uses [Preferences.jl](https://github.com/JuliaPackaging/Preferences.jl) to
> allow the user to choose which MPI implementation to use for a project. This provides
> a single source of truth that can be used for JLL packages (Julia packages providing C libraries)
> that link against MPI, localizes the choice of MPI implementation to a project.

Reviewer: ```suggestion
The MPIPreferences.jl package allows the user to choose which MPI implementation to use in MPI.jl. It uses [Preferences.jl](https://github.com/JuliaPackaging/Preferences.jl) to
configure the MPI backend for each project separately. This provides
a single source of truth that can be used for JLL packages (Julia packages providing C libraries)
that link against MPI.
```
I tried to simplify the paragraph somewhat and also to avoid starting 4 consecutive sentences with "this". This is just a suggestion of course; feel free to modify or discard.

- [Fujitsu MPI](https://www.fujitsu.com/global/about/resources/publications/technicalreview/2020-03/article07.html#cap-03)
- [HPE MPT/HMPT](https://support.hpe.com/hpesc/public/docDisplay?docLocale=en_US&docId=a00105727en_us)

### Configuration

Run `MPIPreferences.use_system_binary()`. This will attempt to locate and to identify any available MPI implementation, and create a file called `LocalPreferences.toml` adjacent to the current `Project.toml`.

```sh
julia --project -e 'using MPI; MPIPreferences.use_system_binary()'
```

If the implementation is changed, you will need to call this function again. See the [`MPIPreferences.use_system_binary`](@ref) documentation for specific options.

!!! note
    You can copy `LocalPreferences.toml` to a different project folder, but you must list
    `MPIPreferences` in the `[extras]` section of the `Project.toml` for the settings
    to take effect. Due to a bug in Julia (until `v1.6.5` and `v1.7.1`), getting preferences
    from transitive dependencies is broken (https://github.com/JuliaPackaging/Preferences.jl/issues/24).

>     To fix this update your version of Julia, or add `MPIPreferences` as a direct dependency to your project.

Reviewer 2: This note was here also before, but I suspect as it's phrased now it may be a bit confusing.  As far as I understand, `MPIPreferences` should be in the active environment, which *for running tests* means that it should be in `[extras]`, right?  This sounds like it should *always* be in `[extras]`.  Or am I getting it wrong?

Author: I think it also works if it is in the `[deps]` section?

Author: I've updated this, let me know what you think.

Reviewer 2: Yes, I think it's better now, thanks!

- [Fujitsu MPI](https://www.fujitsu.com/global/about/resources/publications/technicalreview/2020-03/article07.html#cap-03)
- [HPE MPT/HMPT](https://support.hpe.com/hpesc/public/docDisplay?docLocale=en_US&docId=a00105727en_us)

> ### Configuration

Reviewer: While not strictly necessary, I propose to make it clear that users need to have MPIPreferences.jl installed separately.

~~~suggestion
Furthermore, you need to install the MPIPreferences.jl package, e.g., by executing
```shell
julia -e 'using Pkg; Pkg.add("MPIPreferences")'
```

### Configuration
~~~

Author: Thanks, I added this further up.

```sh
julia --project -e 'using MPIPreferences; MPIPreferences.use_system_binary()'
```

If the implementation is changed, you will need to call this function again. See the [`MPIPreferences.use_system_binary`](@ref) documentation for specific options.

!!! note
    You can copy `LocalPreferences.toml` to a different project folder, but you must list
    `MPIPreferences` in the `[extras]` or `[deps]` section of the `Project.toml` for the settings
    to take effect.

!!! note
    Due to a bug in Julia (until `v1.6.5` and `v1.7.1`), getting preferences
    from transitive dependencies is broken (https://github.com/JuliaPackaging/Preferences.jl/issues/24).
    To fix this update your version of Julia, or add `MPIPreferences` as a direct dependency to your project.

### Notes to HPC cluster adminstators

Preferences are merged across the Julia load path, such that it is feasible to provide a module file that appends a path to
`JULIA_LOAD_PATH` variable that contains system-wide preferences. The steps are as follows:

> 1. Run [`MPIPreferences.use_system_binary()`](@ref MPIPreferences.use_system_binary), which will generate a file `LocalPreferences.toml` containing something like the following:

Reviewer 2: Note that the numbered list always restarts from 1: https://juliaparallel.org/MPI.jl/previews/PR594/configuration/#Notes-to-HPC-cluster-adminstators I believe you have to indent all blocks within each entry with two spaces, including the code blocks.

Author: Thanks

Reviewer 2: Sorry, it was three spaces, not three, I pushed directly a fix, it was too complicated to make a suggestion here, but now it renders correctly for me locally.

Author: Thank you!
